2.0/m_requirectcp.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *       Copyright (C) Nikos `UrL` Papakonstantinou <url.euro@gmail.com>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.      If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "inspircd.h"
+/* $ModConfig: <requirectcp ctcp="VERSION" accepted="Howdy buddy,you are authorized to use this server!" declined="You have been blocked!Please get a better client."> */
+/* $ModDesc: Blocks clients not replying to CTCP like botnets/spambots/floodbots. */
+/* $ModAuthor: Nikos `UrL` Papakonstantinou */
+/* $ModAuthorMail: url.euro@gmail.com */
+/* $ModDepends: core 2.0 */
+
+/*
+ * Supports connect class param & Sno(S).
+ * Add requirectcp="true" which enables or disables the module on that port(s).
+ */
+
+class ModuleRequireCTCP : public Module
+{
+	LocalIntExt ext;
+	std::string accepted;
+	std::string declined;
+	std::string ctcp;
+ public:
+	ModuleRequireCTCP()
+	: ext("ctcptime_wait", this)
+	{
+	}
+
+	void init()
+	{
+		ServerInstance->Modules->AddService(ext);
+		Implementation eventlist[] = { I_OnUserRegister, I_OnPreCommand, I_OnCheckReady, I_OnRehash };
+		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
+		OnRehash(NULL);
+	}
+
+	Version GetVersion()
+	{
+		return Version("Blocks clients not replying to CTCP.");
+	}
+
+	void OnRehash(User* user)
+	{
+		ConfigTag* tag = ServerInstance->Config->ConfValue("requirectcp");
+		ctcp = tag->getString("ctcp", "VERSION");
+		declined = tag->getString("declined", "You have been blocked!Please get a better client.");
+		accepted = tag->getString("accepted", "Howdy buddy,you are authorized to use this server!");
+	}
+
+	ModResult OnUserRegister(LocalUser* user)
+	{
+		ConfigTag* tag = user->MyClass->config;
+		if (tag->getBool("requirectcp", true))
```

Comment:
I think it's set to false by default. Add requirectcp=true inside the connect class to enable it otherwise it returns MOD_RES_PASSTHRU.